### ISSUE TITLE
atproto/common-web package in NPM is missing utf8ToB64Url() and b64UrlToUtf8()

### ISSUE THREAD

**Issue by rudyfraser** [bluesky-social/atproto#1136]
**Describe the bug**

NPM registry for [atproto/common-web](https://www.npmjs.com/package/@atproto/common-web?activeTab=code) strings.ts is missing functions utf8ToB64Url and b64UrlToUtf8 (see [here](https://github.com/bluesky-social/atproto/blob/bf4ac79cef9ce36113ac9daae65c1315983ee263/packages/common-web/src/strings.ts#L15-L21))
This causes the verifyJwt() function in atproto/xrpc-server to throw "poorly formatted jwt" errors [because of these lines](https://github.com/bluesky-social/atproto/blob/bf4ac79cef9ce36113ac9daae65c1315983ee263/packages/xrpc-server/src/auth.ts#L91-L105).

**To Reproduce**

1. Clone the feed-generator example code
2. Install dependencies
3. Uncomment the validateAuth() lines in the example code ([here](https://github.com/bluesky-social/feed-generator/blob/e70ea59066fed9360aa0c1f0891767ce5b119efb/src/methods/feed-generation.ts#L22-L30))
4. Run yarn start
5. Make a GET request to your algo endpoint with a valid bearer token
6. You should get a 401 error with response 'BadJwt'

**Expected behavior**

I would expect that the auth result would pass and a success response would be returned
**Details**

 - Operating system: macOS 13.0.1
 - Node version: v19.4.0

**Additional context**

**Comment by rudyfraser** [comment]
Related issue --[ this parameter ](https://github.com/bluesky-social/atproto/blob/bf4ac79cef9ce36113ac9daae65c1315983ee263/packages/crypto/src/types.ts#L2)doesn't exist in the crypto package from the registry

**Comment by CooperEdmunds** [comment]
Hey @rudyfraser - I'm also running into this. Did you get to the bottom of it?

**Comment by rudyfraser** [comment]
@CooperEdmunds no, I started recreating the feed-generator in Rust since I had intended to for a while. There are also working examples in Python.
I found the NPM package was too out-of-date with many parts needing updating.

**Comment by dholms** [comment]
Huh that's a bit surprising - we bundles our packages before publishing so I would expect that missing functions from a dependency would not be causing this 🤔

I would also expect a different error if those functions were not available - a 500 from a runtime error instead of a 401

How are you generating your JWT? My guess is that the JWT you're generating does not match expectations.

That being said, these packages deserve a republish, so I'll get to that here soon

**Comment by dholms** [comment]
Hey @jetthollister thanks for the full report that helps me get a sense of what's going on!

I'll update these packages today, but I don't expect it will fix the issue here.

1. You'll notice that `@atproto/xrpc-server` package.json points to `dist/index.js` not the TS code. `dist/index.js` is a fully bundled package that should not rely on the outdated `@atproto/common-web` package. If this missing dependency was in the code path, you would get compile errors on the server or runtime errors when it hit the missing code path

2. It looks like you're using an access token for an account on our PDS. This isn't the right token to use for the feed generator. 

There are two different types of JWTs floating around:
- account auth: these are normal looking jwt access/refresh tokens that you might encounter from something like auth0. You can read more about it [here](https://atproto.com/specs/xrpc#inter-service-authentication-temporary-specification)
- service <> service auth: these are jwts that a _service_ passes to another service to authenticate a request _on behalf of_ a user. You can read more about them [here](https://atproto.com/specs/xrpc#inter-service-authentication-temporary-specification)

(yes we know this is a bit confusing, we're planning a larger revamp to our auth system but haven't had the bandwidth for it yet)

The token that you're passing in that is failing account validation is an account auth token. While the validating code is looking for a service auth token.

The best way to get a production service auth token is to proxy a `getFeedSkeleton` request to your server through our PDS - which can be done after launching the feed.
